naming/src/main/resources/schema/jboss-as-naming_3_0.xsd

```diff
@@ -0,0 +1,239 @@
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="module" type="xs:token" use="required">
+            <xs:annotation>
+                <xs:documentation>
+                    The module to load the JNDI entry from
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="class" type="xs:token" use="required">
+            <xs:annotation>
+                <xs:documentation>
+                    The javax.naming.spi.ObjectFactory that provides the value.
+                    Note that when getObjectInstance is invoked the first Object parameter
+                    will be equal to the JNDI name of this binding. The second and third
+                    parameters will always be null, while the last one, the environment, may
+                    be populated using the environment type child element.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+    </xs:complexType>
+
+
+    <xs:complexType name="externalContextType">
+        <xs:sequence>
+        	<xs:element name="environment" type="environmentType" minOccurs="0" maxOccurs="1"/>
+        </xs:sequence>
+        <xs:attribute name="name" type="xs:token" use="required">
+            <xs:annotation>
+                <xs:documentation>
+                    The JNDI name to bind to
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="module" type="xs:token" use="required">
+            <xs:annotation>
+                <xs:documentation>
+                    The module to load the external context from. If this is not specified then
+                    the naming module will be used.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="class" type="xs:token" use="required">
+            <xs:annotation>
+                <xs:documentation>
+                    The type of the context to bind
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="cache" type="xs:boolean" use="optional">
+            <xs:annotation>
+                <xs:documentation>
+                    If this is true them the resulting context will be cached and reused.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+    </xs:complexType>
+
+	<xs:complexType name="environmentType">
+		<xs:annotation>
+			<xs:documentation>
+                <![CDATA[
+               Enclosing element for a list of environment properties.
+             ]]>
+			</xs:documentation>
+		</xs:annotation>
+		<xs:sequence>
+			<xs:element name="property" minOccurs="0" maxOccurs="unbounded"
+				type="propertyType" />
+		</xs:sequence>
+	</xs:complexType>
+
+	<xs:complexType name="propertyType">
```

Comment:
How about _environmentPropertyType_ instead of _propertyType_ ?

Reply:
Sorry, I didn't understand why environmentPropertyType?

Reply:
I meant to distinguish it as a type of environment element. It was just a proposal, you don't need to follow it.

Reply:
We have only one propertyType, so I think it is fine to have this one.

Reply:
ok